### PR TITLE
[Refactor] Introduce a new sharable drawer list view model 

### PR DESCRIPTION
**Pull request by JoshuaLai**
## Purpose
Introduce a new `DrawerListItemViewModel` and `SelectableDrawerListItemViewModel` to be shared for any list items
Removed `LeaveCallConfirmationViewModel`
Removed `MoreCallOptionsListCellViewModel`
Removed `AudioDevicesListCellViewModel` 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [x] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [x] View Data Injection
- [ ] Demo App UI/UX update

### CODE REVIEW

**Comment by jpeng-ms** [comment]
did you manage to resolve the memory leak issue in this new approach?

**Comment by JoshuaLai** [comment]
> did you manage to resolve the memory leak issue in this new approach?
This PR isn't meant to be a bug fix but more of a refactor to clean up view models with duplicated roles and responsibilities.

**Review by vhuseinova-msft — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/FluentUI/Wrapper/AudioDevicesListViewController.swift [hunk 1]
```diff
@@ -50,7 +50,7 @@ extension AudioDevicesListViewController: UITableViewDataSource, UITableViewDele
     }
 
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        self.items[indexPath.row].switchAudioDevice()
+        self.items[indexPath.row].action()
```
Comment (palatter) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/FluentUI/Wrapper/AudioDevicesListViewController.swift:53]: Could we comment on this or put it in seperate method to indicate that the action is to switch Audio device?

**Review by emlynmac — DISMISSED** [review]

**Review by palatter — DISMISSED** [review]
Looks good, just one minor comment.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/FluentUI/Wrapper/AudioDevicesListViewController.swift (hunk 1 above)
Comment (JoshuaLai) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/FluentUI/Wrapper/AudioDevicesListViewController.swift]: Added a new helper method to be more clear

**Review by emlynmac — APPROVED** [review]

**Review by vhuseinova-msft — APPROVED** [review]

**Pull request merged**